### DISCUSSION TITLE
Is the list of file content types intended as a suggestion or a requirement

### DISCUSSION THREAD

**Discussion by nmayle** [CMSgov/price-transparency-guide#58] [Q&A] [answered]
There is a list of open format file content types on the readme page: JSON, XML, YAML.  Is this list intended to define the only file types that can be used or is it just a suggestion for a few valid types, but others are also valid?

For example, can payers use CSV files?  

If  other types are allowed, will the details of how to organize the data in those file types be defined?

**Comment by ghost** [comment]
Based on the open formats listed, "they" appear to be looking for a key-value pair type representation, as found in json and yaml while xml is a bit wordy but conceptually the same, letting the file data to be readable and mappable to domain objects . 

Your example of using CSV would be an interesting exercise to produce a in-network-rates file based on the schema definitions that contain Data Types consisting of arrays of value and objects.

**Comment by shaselton-usds** [comment, marked as answer]
From the final rule, "all machine-readable files must conform to a non-proprietary, open-standards format". 

This was meant to be flexible given that each organization producing the files have different technology and levels of maturity. 

Some formats will lend themselves to be a more natural selection than others. This data has parent-child relationships of which can be difficult (not to mention, quite in-efficient and redundant) to represent in a CSV format.  JSON is the recommended format from the department. 

We can work to develop alternative formats, such as CSV, but that wouldn't be the top priority at the moment.
